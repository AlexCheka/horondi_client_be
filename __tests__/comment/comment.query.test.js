/* eslint-disable no-undef */
const { gql } = require('@apollo/client');
require('dotenv').config();
const { COMMENT_NOT_FOUND } = require('../../error-messages/comment.messages');
const { setupApp } = require('../helper-functions');

const {
  newComment,
  validEmail,
  invalidEmail,
  productId,
  wrongData,
} = require('./comment.variables');

let commentId = '';
let operations;
describe('Comment queries', () => {
  beforeAll(async () => {
    operations = await setupApp();
    const res = await operations
      .mutate({
        mutation: gql`
          mutation($productId: ID!, $comment: commentInput!) {
            addComment(productId: $productId, comment: $comment) {
              ... on Comment {
                _id
              }
              ... on Error {
                message
              }
            }
          }
        `,
        variables: { productId: productId, comment: newComment },
      })
      .catch(e => e);
    commentId = res.data.addComment._id;
  });

  afterAll(async () => {
    await operations
      .mutate({
        mutation: gql`
          mutation($id: ID!) {
            deleteComment(id: $id) {
              ... on Comment {
                _id
              }
              ... on Error {
                statusCode
                message
              }
            }
          }
        `,
        variables: { id: commentId },
      })
      .catch(e => e);
  });

  it(' Should receive all comments writen by selected user', async () => {
    const res = await operations
      .query({
        variables: {
          userEmail: validEmail,
        },
        query: gql`
          query($userEmail: String!) {
            getAllCommentsByUser(userEmail: $userEmail) {
              text

              product {
                _id
              }
              show
              user {
                email
              }
            }
          }
        `,
      })
      .catch(e => e);
    console.log(res);

    expect(res.data.getAllCommentsByUser).toMatchSnapshot();
    expect(res.data.getAllCommentsByUser).toBeDefined();
    expect(res.data.getAllCommentsByUser).toContainEqual({
      product: { _id: productId },
      text: newComment.text,
      user: newComment.user,
      show: newComment.show,
    });
  });

  // it(' should return error messagePassing unexisting email ', async () => {
  //   const res = await operations.query({
  //     variables: {
  //       userEmail: invalidEmail,
  //     },
  //     query: gql`
  //       query($userEmail: String!) {
  //         getAllCommentsByUser(userEmail: $userEmail) {
  //           text
  //           date
  //           product {
  //             _id
  //           }
  //           show
  //         }
  //       }
  //     `,
  //   });
  //   console.log('second test', res);
  //   expect(res.data.getAllCommentsByUser).toBeDefined();
  //   expect(res.data.getAllCommentsByUser).toHaveProperty('statusCode', 404);
  //   expect(res.data.getAllCommentsByUser).toHaveProperty(
  //     'message',
  //     COMMENT_NOT_FOUND
  //   );
  // });

  // it(' should return error message Passing not email string ', async () => {
  //   const res = await operations.query({
  //     variables: {
  //       userEmail: wrongData,
  //     },
  //     query: gql`
  //       query($userEmail: String!) {
  //         getAllCommentsByUser(userEmail: $userEmail) {
  //           text
  //           date
  //           product {
  //             _id
  //           }
  //           show
  //         }
  //       }
  //     `,
  //   });
  //   console.log('third test', res);
  //   expect(res.data.getAllCommentsByUser).toBeDefined();
  //   expect(res.data.getAllCommentsByUser).toHaveProperty('statusCode', 404);
  //   expect(res.data.getAllCommentsByUser).toHaveProperty(
  //     'message',
  //     COMMENT_NOT_FOUND
  //   );
  // });
});
