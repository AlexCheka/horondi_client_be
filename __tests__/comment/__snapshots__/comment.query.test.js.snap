// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment queries  Should receive all comments writen by selected user 1`] = `
Array [
  Object {
    "product": Object {
      "_id": "be89b5471126f0fdef739755",
    },
    "show": false,
    "text": "Test text",
    "user": Object {
      "email": "ermn7dyptp@yahoo.com",
    },
  },
]
`;
